### PR TITLE
em-801 have 60% branch coverage for unit tests

### PR DESCRIPTION
**Pull request by btzc**
- Refactor code to expose certain parts for easier testing
- Have 60% branch coverage

### CODE REVIEW

File: src/app/projects/project-detail/documents/documents-tab-content.component.spec.ts [hunk 1]
```diff
@@ -37,10 +37,28 @@ describe('DocumentsTabContentComponent', () => {
...
+      expect(component.project).toBeTruthy;
+    });
+  });
+  describe('parseData(data)', () => {
+    beforeEach(() => {
+      const data = { project: new Project() };
+      data.project.collections = new CollectionsList();
+      component.parseData(data);
+    });
+    it('should return data this.project', () => {
```
Comment (NickPhura) [inline src/app/projects/project-detail/documents/documents-tab-content.component.spec.ts]: nit: 'should return data for this.project'

File: src/app/services/project.service.spec.ts [hunk 2]
```diff
@@ -1,23 +1,473 @@
...
+  }
+
+  function createResponseItem(response: any) {
+    responseItem = {
+      response: response
+    };
+    return responseItem;
+  }
+
+  function creategGetAllMockresponses(projectObjs: any) {
```
Comment (NickPhura) [inline src/app/services/project.service.spec.ts]: spelling: creategGet

File: src/app/authorizations/authorizations.component.spec.ts [hunk 3]
```diff
@@ -1,30 +1,53 @@
...
 import { Ng2PageScrollModule } from 'ng2-page-scroll';
-import { RouterTestingModule } from '@angular/router/testing';
+import { Router, NavigationEnd } from '@angular/router';
 
 import { AuthorizationsComponent } from './authorizations.component';
 
 describe('AuthorizationsComponent', () => {
   let component: AuthorizationsComponent;
   let fixture: ComponentFixture<AuthorizationsComponent>;
+  let ni;
```
[the remark this answers is not in the document]
Comment (btzc) [reply src/app/authorizations/authorizations.component.spec.ts]: done

File: src/app/filters/order-by.pipe.spec.ts [hunk 4]
```diff
@@ -1,8 +1,347 @@
...
+    });
+    it('returns records', () => {
+      records = objectArrayFactory(10);
+      args = argsObjectFactory( 'dateAdded', null );
+      expectedResponse = records;
+
+      expect(pipe.transform(records, args).length).toBe(expectedResponse.length);
+    });
+  });
+  describe('given a property of arg is null', () => {
```
[the remark this answers is not in the document]
Comment (btzc) [reply src/app/filters/order-by.pipe.spec.ts]: removed

File: src/app/projects/project-detail/documents/documents-tab-content.component.spec.ts (hunk 1 above)
Comment (btzc) [reply src/app/projects/project-detail/documents/documents-tab-content.component.spec.ts]: done

File: src/app/services/project.service.spec.ts (hunk 2 above)
Comment (btzc) [reply src/app/services/project.service.spec.ts]: fixed

**Comment by NickPhura** [comment]
👍

**Pull request merged**